lib/msf/ui/console/command_dispatcher/db/klist.rb

```diff
@@ -118,4 +134,65 @@ def cmd_klist(*args)
   def kerberos_ticket_storage
     @kerberos_ticket_storage ||= Msf::Exploit::Remote::Kerberos::Ticket::Storage::ReadWrite.new(framework: framework)
   end
+
+  private
+
+  def ticket_status(ticket)
+    if ticket.expired?
+      '>>expired<<'
+    elsif ticket.status.blank?
+      'active'
+    else
+      ticket.status
+    end
+  end
+
+  # Sets the status of the tickets
+  #
+  # @param [Symbol] mode The status (:activate or :deactivate) to apply to the ticket(s)
+  # @param [Array<StoredTicket>] tickets The tickets which statuses are to be updated
+  # @return [Integer] Number of tickets updated
+  def set_activation_status(mode, tickets)
+    result = []
+    if mode == :activate
+      result = kerberos_ticket_storage.activate_ccache(ids: tickets.map(&:id))
+    elsif mode == :deactivate
+      result = kerberos_ticket_storage.deactivate_ccache(ids: tickets.map(&:id))
+    end
+    result.size
+  end
+
+  def ticket_search(host_ranges, id_search)
+    ticket_results = []
+
+    if id_search
+      id_search.each do |id|
```

Comment:
Let's add a comment here that we're iterating over the IDs manually as the remote database service doesn't support bulk ID searches